templates/default/elasticsearch.yml.erb

```diff
@@ -0,0 +1,159 @@
+######################### ElasticSearch Configuration  ########################
+
+# This file is managed by Chef, do not edit manually, your changes *will* be overwritten!
+#
+# Please see the source file for context and more information:
+#
+# https://github.com/elasticsearch/elasticsearch/blob/master/config/elasticsearch.yml
+#
+# To set configurations not exposed by this template, set the
+# `@config[:custom_config]` attribute in your node configuration,
+# `elasticsearch/settings` data bag, role/environment definition, etc:
+#
+#     // ...
+#     'threadpool.index.type' => 'fixed',
+#     'threadpool.index.size' => '2'
+#     // ...
+
+################################### Cluster ###################################
```

Comment:
> Can we base the template on the following pruned version: https://github.com/elastic/elasticsearch/blob/trim-config/config/elasticsearch.yml ?

Sure!

> Also, since we support putting any key:value pair from @config file, we don't need all those lines? It will be included in the file automatically?

That's also correct. Anything we want formatted in any fancy way, we can template out, otherwise the lines will just go in as-is :)